### PR TITLE
Seed noniid permutation tests

### PR DESCRIPTION
**Pull request by elisno**
This PR adds a `seed` argument to the constructor of `NonIIDIssueManager` that can be used to reproduce the results of the permutation tests run when calling the `find_issues` method of the class.

### CODE REVIEW

File: cleanlab/datalab/issue_manager/noniid.py [hunk 1]
```diff
@@ -103,6 +103,7 @@ def __init__(
         metric: Optional[str] = None,
         k: int = 10,
         num_permutations: int = 25,
+        seed: Optional[int] = None,
```
Comment (jwmueller) [inline cleanlab/datalab/issue_manager/noniid.py:106]: why choose default None? Don't we want this to be deterministic by default?
We should still allow user to pass None if they dont want to set seed, but my suggestion is to set default = 0 say

Comment (jwmueller) [inline cleanlab/datalab/issue_manager/noniid.py:106]:
```suggestion
        seed: Optional[int] = 0,
```

File: cleanlab/datalab/issue_manager/noniid.py [hunk 2]
```diff
@@ -239,6 +241,8 @@ def _build_statistics_dictionary(self, knn_graph: csr_matrix) -> Dict[str, Dict[
     def _permutation_test(self, num_permutations) -> float:
         N = self.N
 
+        if self.seed is not None:
+            np.random.seed(self.seed)
```
Comment (jwmueller) [inline cleanlab/datalab/issue_manager/noniid.py:245]: @jecummin can you confirm all non-determinism in this method happens after here and is only in np methods?

**Comment by elisno** [comment]
There's flaky unit test that's unrelated to this PR that's failing CI.

It's addressed in https://github.com/cleanlab/cleanlab/pull/695.

File: cleanlab/datalab/issue_manager/noniid.py (hunk 1 above)
Comment (elisno) [reply cleanlab/datalab/issue_manager/noniid.py]: Changed behavior to be deterministic by default.

File: cleanlab/datalab/issue_manager/noniid.py (hunk 2 above)
Comment (jecummin) [reply cleanlab/datalab/issue_manager/noniid.py:245]: Confirmed!

**Review by jwmueller — APPROVED** [review]
Thanks for this!

**Pull request merged**